src/test/java/org/terasology/launcher/repositories/JenkinsRepositoryAdapterTest.java

```diff
@@ -0,0 +1,205 @@
+                "        },\n" +
+                "        {\n" +
+                "          \"fileName\": \"versionInfo.properties\",\n" +
+                "          \"relativePath\": \"distros/omega/versionInfo.properties\"\n" +
+                "        }\n" +
+                "      ],\n" +
+                "      \"number\": 1,\n" +
+                "      \"result\": \"SUCCESS\",\n" +
+                "      \"timestamp\": 1604285977306,\n" +
+                "      \"url\": \"http://jenkins.terasology.io/teraorg/job/Nanoware/job/Omega/job/develop/1/\"\n" +
+                "    }\n" +
+                "  ]\n" +
+                "}";
+    }
+
+    static String nullArtifactsPayload() {
+        return "{ \n" +
+                "  \"builds\": [\n" +
+                "    {\n" +
+                "      \"number\": 1, \"result\": \"SUCCESS\", \"timestamp\": 1604285977306, \n" +
+                "      \"url\": \"http://jenkins.terasology.io/teraorg/job/Nanoware/job/Omega/job/develop/1/\"\n" +
+                "    }\n" +
+                "  ]\n" +
+                "}";
+    }
+
+    static String emptyArtifactsPayload() {
+        return "{\n" +
+                "  \"builds\": [\n" +
+                "    {\n" +
+                "      \"artifacts\": [],\n" +
+                "      \"number\": 1,\n" +
+                "      \"result\": \"SUCCESS\",\n" +
+                "      \"timestamp\": 1604285977306,\n" +
+                "      \"url\": \"http://jenkins.terasology.io/teraorg/job/Nanoware/job/Omega/job/develop/1/\"\n" +
+                "    }\n" +
+                "  ]\n" +
+                "}";
+    }
+
+    static String incompleteArtifactsPayload() {
+        return "{\n" +
+                "  \"builds\": [\n" +
+                "    {\n" +
+                "      \"artifacts\": [\n" +
+                "        {\n" +
+                "          \"fileName\": \"versionInfo.properties\",\n" +
+                "          \"relativePath\": \"distros/omega/versionInfo.properties\"\n" +
+                "        }\n" +
+                "      ],\n" +
+                "      \"number\": 1,\n" +
+                "      \"result\": \"SUCCESS\",\n" +
+                "      \"timestamp\": 1604285977306,\n" +
+                "      \"url\": \"http://jenkins.terasology.io/teraorg/job/Nanoware/job/Omega/job/develop/1/\"\n" +
+                "    }\n" +
+                "  ]\n" +
+                "}";
+    }
+
+    static List<String> incompletePayloads() {
+        return List.of(
+                "{}",
+                "{ \"builds\": [] }",
+                nullArtifactsPayload(),
+                emptyArtifactsPayload(),
+                incompleteArtifactsPayload()
+        );
+    }
+
+    static Stream<Arguments> incompleteResults() {
+        return incompleteResults.stream().map(Arguments::of);
+    }
+
+    @Test
+    @DisplayName("handle null Jenkins response gracefully")
+    void shouldHandleNullJenkinsResponseGracefully() {
```

Comment:
```suggestion
    void handleNullJenkinsResponseGracefully() {
```

same for following occurrences